src/ingest-pipeline/md/data_collection_types/epic_metadata_data_collection.py

```diff
@@ -0,0 +1,122 @@
+            if not any(glob.iglob(os.path.join(path, match.format(offsetdir=offsetdir)))):
+                print("not found!")
+                return False
+        return True
+
+    def __init__(self, path):
+        """
+        path is the top level directory of the collection
+        """
+        super().__init__(path)
+        self.offsetdir = self.find_top(self.topdir, self.top_target, self.dir_regex)
+        assert self.offsetdir is not None, "Wrong dataset type?"
+
+    def collect_metadata(self, component=None, component_process=None):
+        md_type_tbl = self.get_md_type_tbl()
+        rslt = {}
+        cl = []
+        for match, md_type in self.expected_files + self.optional_files:
+            print("collect match %s" % match.format(offsetdir=self.offsetdir))
+            for fpath in glob.iglob(
+                os.path.join(self.topdir, match.format(offsetdir=self.offsetdir))
+            ):
+                print("collect from path %s" % fpath)
+                this_md = md_type_tbl[md_type](fpath).collect_metadata()
+                if this_md is not None:
+                    rslt[os.path.relpath(fpath, self.topdir)] = this_md
+                    fname = os.path.basename(fpath)
+                    if "metadata" in fname and fname.endswith(".tsv"):
+                        assert isinstance(this_md, list), "metadata.tsv did not produce a list"
+                        rec_list = this_md
+                        for rec in rec_list:
+                            assert "derived_dataset_type" in rec, ("No derived_dataset_type found "
+                                                                   "in metadata.tsv")
+                            for key in ["data_path", "contributors_path"]:
+                                assert (
+                                    key in rec
+                                ), 'metadata.tsv does not have a "{}" column'.format(key)
+                            this_dict = {"metadata": rec}
+                            for sub_key, dict_key in [
+                                ("contributors_path", "contributors"),
+                                ("antibodies_path", "antibodies"),
+                            ]:
+                                if sub_key in rec:
+                                    assert rec[sub_key].endswith(
+                                        ".tsv"
+                                    ), 'TSV file expected, received "{}"'.format(rec[sub_key])
+                                    sub_path = os.path.join(os.path.dirname(fpath), rec[sub_key])
+                                    sub_parser = md_type_tbl["TSV"](sub_path)
+                                    sub_md = sub_parser.collect_metadata()
+                                    this_dict[dict_key] = sub_md
+                            cl.append(this_dict)
+
+        rslt["components"] = cl
+        rslt["collectiontype"] = "epiic_metadatatsv"
```

Comment:
@sunset666 - minor fix here to rename to epic_metadatatsv

Reply:
good catch.